oeps/oep-38-Data-Modeling.rst

```diff
@@ -0,0 +1,438 @@
+needed to support the business processes within the scope of
+corresponding information systems in organizations.
+
+Data Modeling Standards - The most basic and standard design principles
+for data modeling. These standards must be adhered to when creating new
+models or updating existing models in the edX ecosystem.
+
+IDA - “Independently Deployed Application”. Similar to a microservice or
+stand alone application.
+
+OpenedX Ecosystem
+
+------------------
+
+Used in this document, the phrase "OpenedX Ecosystem" includes the
+services that comprise and support the running of an OpenedX
+installation (e.g.,. edx.org). This includes:
+
+- The LMS and Studio
+
+- IDAs such as Notes, Ecommerce, and Forums
+
+- Backend systems such as data analytics and operational logging
+
+- Third party services used to extend functionality of applications,
+such as Optimizely, Google Analytics, and Sailthru
+
+Specification
+
+=============
+
+Guiding Principles
+------------------
+
+Data is an asset
+~~~~~~~~~~~~~~~~
+
+edX uses data to decide which people should receive marketing emails,
+who passes or fails a course, or how much to pay our partners. Our
+partners use our data to target users who fit certain criteria and to
+refine and improve their courses. The data we collect today may be used
+to gain new educational insights we have yet to realize. Decision making
+at edX should be data-driven and based on this collected data.
+
+This data is one of our most valuable assets and it should be and it
+should be a first-order concern. Save everything (disk is cheap). Fire
+lots of events. The more data we are able to collect about users,
+behavior, and system state now, the more opportunities we’ll have to
+improve our decision-making in the future.
+
+Use industry best practices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Data stored in the OpenEdX ecosystem should adhere to industry best
+practices. For example, since an industry best practice uses numerical
+identifiers to identify rows of data, the edX ecosystem should also use
+numerical identifiers. Adhering to industry best practices and the
+practices outlined in this document ensures our data is approachable for
+experienced engineers and new hires alike and allows us to leverage 3rd
+party tools more easily to assist in the analysis phase.
+
+Don’t make me think
+~~~~~~~~~~~~~~~~~~~
+
+Steve Krug, the author of “Don’t Make Me Think”, says: “Your objective
+should always be to eliminate instructions entirely by making everything
+self-explanatory, or as close to it as possible.” This principal should
+also be applied to data modeling. Data models should have descriptive
+names clearly identifying the data that the model holds. The field names
+should make sense to most people familiar with the domain, without
+needing much clarifying documentation.
+
+Prioritize ease of analysis
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For every hour of engineer-time spent creating a data model, several
```

Comment:
"several thousand" strikes me as hyperbolistic; how about "many more"?

Reply:
There was a lot of back and forth on this line before I published the OEP. I agree that it turned out a bit hyperbolic. While I am unable to back up this fact with exact numbers I'm not sure that we should try. I like your suggestion @kdmccormick.

@mulby I do not think this section suggests that models should be optimized for analysis. I do however believe that many more hours are spent analyzing the data. A little more thought about the analysis up front could save us a lot of time later.

Reply:
Ok I take that back I do see I said Prioritize. How about consider?